src/fillTank.test.js

```diff
@@ -1,11 +1,98 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+   const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should be declared ', () => {
+    expect(fillTank).toBeInstanceOf(Function);
+  });
+
+  it('should calculate full tank If the amount is not given', () => {
+    const customer = {
+      money: 3000, // customer account balance
+      vehicle: {
+        maxTankCapacity: 40, // fuel tank volume
+        fuelRemains: 0, // Remaining fuel in the tank
+      },
+    };
+
+    fillTank(customer, 30);
+
+    expect(customer.money).toBe(1800);
+    expect(customer.vehicle.fuelRemains).toBe(40);
+  });
+
+  it('should calculate full tank If the amount is greater than the tank can accommodate', () => {
+    const customer = {
+      money: 3000, // customer account balance
+      vehicle: {
+        maxTankCapacity: 20, // fuel tank volume
+        fuelRemains: 5, // Remaining fuel in the tank
+      },
+    };
+
+    fillTank(customer, 50, 20);
+
+    expect(customer.money).toBe(2250);
+    expect(customer.vehicle.fuelRemains).toBe(20);
+  });
+
+  it('should fill the fuel if the client can pay for it', () => {
+    const customer = {
+      money: 375, // customer account balance
+      vehicle: {
+        maxTankCapacity: 30, // fuel tank volume
+        fuelRemains: 5, // Remaining fuel in the tank
+      },
+    };
+
+    fillTank(customer, 25, 20);
 
+    expect(customer.money).toBe(0);
+    expect(customer.vehicle.fuelRemains).toBe(20);
   });
 
-  // write tests here
+  it('should round the poured amount by discarding number to the tenth part.', () => {
```

Comment:
The test description mentions rounding the poured amount by discarding the number to the tenth part. Ensure that the test case verifies this behavior correctly.